mlflow/models/model.py

```diff
@@ -973,6 +977,13 @@ def log_model_metrics_for_step(client, model_id, run_id, step):
                     else None,
                 )
 
+            if model.status in LoggedModelStatus.FINALIZED_STATUSES:
+                raise MlflowException(
+                    f"Model with id {model.model_id} has the status '{model.status}', "
+                    f"so its artifacts cannot be modified.",
+                    BAD_REQUEST,
+                )
```

Comment:
Once a model has been finalized (e.g. because its an external model), we should reject new attempts to modify its artifacts.